src/dbnode/storage/bootstrap/bootstrapper/peers/source.go

```diff
@@ -290,8 +274,16 @@ func (s *peersSource) readData(
 		zap.Int("concurrency", concurrency),
 		zap.Bool("shouldPersist", shouldPersist))
 	if shouldPersist {
-		go s.startPersistenceQueueWorkerLoop(
-			opts, persistenceWorkerDoneCh, persistenceQueue, persistFlush, result, &resultLock)
+		// Spin up persist workers.
+		for i := 0; i < s.opts.ShardPersistenceFlushConcurrency(); i++ {
+			closer, err := s.startPersistenceQueueWorkerLoop(opts,
+				persistWg, persistenceQueue, result, &resultLock)
+			if err != nil {
+				return nil, err
```

Comment:
Should this stop the other workers?

Reply:
The inner workers all take work from the `persistenceQueue` channel which gets closed later `close(persistenceQueue)` (and stops all the workers) and then it waits on these to finish their cleanups with `persistWg.Wait()`.